<a href="https://colab.research.google.com/github/Manu-Gr/Hypothesis-Testing-Q3/blob/main/Hypothesis_Testing_Q3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import chi2_contingency

Assume Null Hypothesis as Ho: Independence of categorical variables (male-female buyer rations are similar across regions 

Thus Alternate Hypothesis as Ha: Dependence of categorical variables (male-female buyer rations are NOT similar across regions 

In [ ]:
data = pd.read_csv('/content/BuyerRatio.csv')
data

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [ ]:
# Make dimensional array
obs=np.array([[50,142,131,70],[435,1523,1356,750]])
obs

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [ ]:
# Chi2 contengency independence test

chi2_contingency(obs) 
# o/p is (Chi2 stats value, p_value, df, expected obsvations)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

Compare P Value with alpha

P value ( 0.66) > alpha (0.05)
Accept Null Hypothesis

We can conclude that male-female buyer rations are NOT similar across regions